### PR TITLE
Try again collecting DB dump

### LINKED ISSUE #2467: Collecting noobaa logs causes internalError for pytest
https://ceph-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/ocs-ci/493/consoleFull

INTERNALERROR> ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc -n openshift-storage --kubeconfig cluster/auth/kubeconfig rsh noobaa-db-0 mongodump --archive=nbcore.gz --gzip --db=nbcore.

### PR DESCRIPTION
**Pull request by ebenahar**
Allow re-collecting of Noobaa DB dump with a limitation of 5 attempts

Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

**Comment by petr-balogh** [comment]
@ebenahar I am now testing this one:  https://github.com/red-hat-storage/ocs-ci/pull/2470/files

I don't think we should only count on one specific issue. What I've read from pytest issue https://github.com/pytest-dev/pytest/issues/333 . When we use hook at the moment I think we should make this more error prone to not see other internal errors. So I covered more things here with try expect to don't have such issue on any other issue we can see.

**Comment by ebenahar** [comment]
@petr-balogh sounds good. Closing

**Comment by ebenahar** [comment]
Reopened in order to have the DB dump collected post MCG test failure to be provided for the investigation of https://bugzilla.redhat.com/show_bug.cgi?id=1838621

File: ocs_ci/ocs/utils.py [hunk 1]
```diff
@@ -830,7 +830,13 @@ def collect_ocs_logs(dir_name, ocp=True, ocs=True, mcg=False):
             '/usr/bin/gather_service_logs worker'
         )
     if mcg:
-        collect_noobaa_db_dump(log_dir_path)
+        mcg_db_collected = False
+        while not mcg_db_collected:
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:834]: Hope this is just for some debugging purpose as I see DNM prefix :) I am adding WIP prefix to title as this really prevent us to merge.

Comment (ebenahar) [reply ocs_ci/ocs/utils.py:834]: +1

**Comment by ebenahar** [comment]
Closing as this PR served its purpose of collecting DB dump post a failure of the MCG test for the investigation of https://bugzilla.redhat.com/show_bug.cgi?id=1838621

**Comment by ebenahar** [comment]
Re-opening in order to allow re-collecting of Noobaa DB dump with a limitation of 5 attempts

File: ocs_ci/ocs/utils.py [hunk 2]
```diff
@@ -830,7 +830,15 @@ def collect_ocs_logs(dir_name, ocp=True, ocs=True, mcg=False):
...
     if mcg:
-        collect_noobaa_db_dump(log_dir_path)
+        counter = 0
+        mcg_db_collected = False
+        while not mcg_db_collected or counter < 5:
+            try:
+                collect_noobaa_db_dump(log_dir_path)
+                mcg_db_collected = True
+            except CommandFailed:
+                counter += 1
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:840]: This ending up in infinite loop @ebenahar .

See this upgrade execution it got stuck in all the time collecting noobaa stuff :(

https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/9782/consoleFull

You should add counter += 1 at the start before line 836.

Comment (petr-balogh) [reply ocs_ci/ocs/utils.py:840]: Opened issue here:
https://github.com/red-hat-storage/ocs-ci/issues/2496#issue-656141735

Comment (petr-balogh) [reply ocs_ci/ocs/utils.py:840]: Fixing here: https://github.com/red-hat-storage/ocs-ci/pull/2497

**Pull request merged**